portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -131,13 +157,56 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty(COMMENT_TIMESTAMP, timestamp);
     commentEntity.setProperty(COMMENT_NAME, name);
     commentEntity.setProperty(COMMENT_LENGTH, text.length());
-    commentEntity.setProperty(COMMENT_EMAIL, email);
     commentEntity.setProperty(COMMENT_NICKNAME, nickname);
+    commentEntity.setProperty(COMMENT_IMAGE_URL, imageUrl);
     
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     // Redirects to the bottom of the current page to see new comment added.
     response.sendRedirect(BOTTOM_OF_PAGE);
   }
+
+  /** 
+   * Returns a URL that points to the uploaded file based on user input in 
+   * {@code formInputElementName}. If the user didn't upload a file, returns an empty Optional.
+   */
+  private Optional<String> getUploadedFileUrl(HttpServletRequest request,
+      String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    Optional<List<BlobKey>> blobKeys = Optional.ofNullable(blobs.get("image"));
+    
+    // Can not get a URL because the user submitted the form without selecting a file, 
+    // so we can't get a URL.
+    if (!blobKeys.isPresent() || blobKeys.get().isEmpty()) {
+      return Optional.empty();
+    }
+
+    // Gets the first index because the comment form only takes in one file input.
+    BlobKey blobKey = blobKeys.get().get(0);
+
+    // Can not get a URL because the user submitted the form on the live server without 
+    // selecting a file, so we can't get a URL. (live server)
```

Comment:
```suggestion
    // Cannot get a URL because the user submitted the form on the live server without 
    // selecting a file, so we can't get a URL. (live server)
```

Reply:
Done!